storage/repository/parse.go

```diff
@@ -43,14 +44,12 @@ func NewFileURI(path string) fyne.URI {
 // Since: 2.0
 func ParseURI(s string) (fyne.URI, error) {
 	// Extract the scheme.
-	scheme := ""
-	for i := 0; i < len(s); i++ {
-		if s[i] == ':' {
-			break
-		}
-		scheme += string(s[i])
+	colonIndex := strings.Index(s, ":")
```

Comment:
This is the incorrect API to use for single characters. It makes more sense to use `strings.IndexByte` in the case.